daml-script/dump/src/main/scala/com/daml/script/dump/Main.scala

```diff
@@ -57,29 +60,68 @@ object Main {
     ()
   }
 
+  private def getACS(
+      client: LedgerClient,
+      parties: List[String],
+      offset: LedgerOffset,
+  )(implicit
+      mat: Materializer
+  ): Future[Map[String, CreatedEvent]] = {
+    val ledgerBegin = LedgerOffset(
+      LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_BEGIN)
+    )
+    if (offset == ledgerBegin) {
+      Future.successful(Map.empty)
```

Comment:
This is purely an optimization right? The code below looks like it’ll work without this.

Reply:
Yes, if the range is going to be empty, then there is no need to issue an API call. The same holds in `getTransactionTrees`. I've added it there as well for consistency.